### PR TITLE
wip: expose simulate1D

### PR DESCRIPTION
**Pull request by lpatiny**

### CODE REVIEW

File: src/signals/spinSystem.ts [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
   couplingConstants: Matrix;
+  /**
+   * Array of chemical shifts of the spin system
+   */
   chemicalShifts: number[];
-  multiplicity: number[];
+  /**
+   * Array of degree of freedom of the spin, it is usually an array of 2.
+   */
+  levels: number[];
```
Comment (lpatiny) [inline src/signals/spinSystem.ts:19]: This should be optional

File: src/signals/spinSystem.ts [hunk 2]
```diff
@@ -1,9 +1,24 @@
 import type { Matrix } from 'ml-matrix';
 
 export interface SpinSystem {
+  /**
+   * isolated clusters from a similarity/connectivity matrix
+   */
   clusters: number[][];
```
Comment (lpatiny) [inline src/signals/spinSystem.ts:7]: This should be optional

File: src/signals/spinSystem.ts [hunk 3]
```diff
@@ -1,9 +1,24 @@
...
   chemicalShifts: number[];
-  multiplicity: number[];
+  /**
+   * Array of degree of freedom of the spin, it is usually an array of 2.
+   */
+  levels: number[];
+  /**
+   * Binary matrix based on the coupling constants.
+   */
   connectivity: Matrix;
```
Comment (lpatiny) [inline src/signals/spinSystem.ts:23]: This should probably not exist but be generated internaly from couplingConstants Matrix

Comment (jobo322) [reply src/signals/spinSystem.ts]: it is generated in the method ensureSpinSystemProps as you did suggest. We could keep this property as optional.

File: src/signals/spinSystem.ts (hunk 1 above)
Comment (jobo322) [reply src/signals/spinSystem.ts]: done

File: src/signals/spinSystem.ts (hunk 2 above)
Comment (jobo322) [reply src/signals/spinSystem.ts]: done

**Pull request merged**